In [1]:
import seaborn as sb
import polars as pl

In [2]:
df = pl.scan_csv("penalties_py_0.txt", separator="\t")

In [3]:
pos_df = df.filter(pl.col("Reaction").str.ends_with("_pos")).select(pl.col("Reaction").str.replace("_pos", ""), pl.selectors.numeric())
neg_df = df.filter(pl.col("Reaction").str.ends_with("_neg")).select(pl.col("Reaction").str.replace("_neg", ""), pl.selectors.numeric())

In [4]:
pos_df.head().collect()

Reaction,WT-d_S141_L007_R1_001,Ob-DHA-e_S154_L007_R1_001,WT-1012-c_S135_L007_R1_001,Ob-DHA-b_S132_L007_R1_001,Ob-EPA-a_S122_L007_R1_001,Ob-c_S136_L007_R1_001,Ob-1012-c_S140_L007_R1_001,WT-EPA-e_S151_L007_R1_001,WT-DHA-d_S143_L007_R1_001,WT-DHA-b_S128_L007_R1_001,WT-EPA-a_S117_L007_R1_001,WT-1012-d_S145_L007_R1_001,Ob-1012-d_S149_L007_R1_001,WT-DHA-a_S118_L007_R1_001,Ob-911-c_S139_L007_R1_001,Ob-d_S146_L007_R1_001,WT-EPA-b_S127_L007_R1_001,Ob-911-a_S124_L007_R1_001,Ob-b_S131_L007_R1_001,WT-1012-b_S130_L007_R1_001,Ob-EPA-c_S137_L007_R1_001,Ob-EPA-e_S153_L007_R1_001,Ob-1012-b_S133_L007_R1_001,WT-911-b_S129_L007_R1_001,WT-911-a_S119_L007_R1_001,WT-DHA-c_S134_L007_R1_001,WT-EPA-d_S142_L007_R1_001,Ob-a_S121_L007_R1_001,Ob-DHA-c_S138_L007_R1_001,WT-1012-a_S120_L007_R1_001,Ob-1012-a_S125_L007_R1_001,Ob-DHA-a_S123_L007_R1_001,WT-e_S150_L007_R1_001,Ob-911-e_S155_L007_R1_001,WT-a_S116_L007_R1_001,Ob-911-d_S148_L007_R1_001,WT-1012-e_S152_L007_R1_001,WT-911-d_S144_L007_R1_001,Ob-DHA-d_S147_L007_R1_001,WT-b_S126_L007_R1_001
str,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64
"""10FTHF5GLUtl""",1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0
"""10FTHF5GLUtm""",1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0
"""10FTHF6GLUtl""",1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0
"""10FTHF6GLUtm""",1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0
"""10FTHF7GLUtl""",1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0


In [5]:
combined_df = pos_df.join(neg_df, on="Reaction", how = "inner", suffix="_neg")

In [6]:
combined_df.head().collect()

Reaction,WT-d_S141_L007_R1_001,Ob-DHA-e_S154_L007_R1_001,WT-1012-c_S135_L007_R1_001,Ob-DHA-b_S132_L007_R1_001,Ob-EPA-a_S122_L007_R1_001,Ob-c_S136_L007_R1_001,Ob-1012-c_S140_L007_R1_001,WT-EPA-e_S151_L007_R1_001,WT-DHA-d_S143_L007_R1_001,WT-DHA-b_S128_L007_R1_001,WT-EPA-a_S117_L007_R1_001,WT-1012-d_S145_L007_R1_001,Ob-1012-d_S149_L007_R1_001,WT-DHA-a_S118_L007_R1_001,Ob-911-c_S139_L007_R1_001,Ob-d_S146_L007_R1_001,WT-EPA-b_S127_L007_R1_001,Ob-911-a_S124_L007_R1_001,Ob-b_S131_L007_R1_001,WT-1012-b_S130_L007_R1_001,Ob-EPA-c_S137_L007_R1_001,Ob-EPA-e_S153_L007_R1_001,Ob-1012-b_S133_L007_R1_001,WT-911-b_S129_L007_R1_001,WT-911-a_S119_L007_R1_001,WT-DHA-c_S134_L007_R1_001,WT-EPA-d_S142_L007_R1_001,Ob-a_S121_L007_R1_001,Ob-DHA-c_S138_L007_R1_001,WT-1012-a_S120_L007_R1_001,Ob-1012-a_S125_L007_R1_001,Ob-DHA-a_S123_L007_R1_001,WT-e_S150_L007_R1_001,Ob-911-e_S155_L007_R1_001,WT-a_S116_L007_R1_001,Ob-911-d_S148_L007_R1_001,…,Ob-DHA-b_S132_L007_R1_001_neg,Ob-EPA-a_S122_L007_R1_001_neg,Ob-c_S136_L007_R1_001_neg,Ob-1012-c_S140_L007_R1_001_neg,WT-EPA-e_S151_L007_R1_001_neg,WT-DHA-d_S143_L007_R1_001_neg,WT-DHA-b_S128_L007_R1_001_neg,WT-EPA-a_S117_L007_R1_001_neg,WT-1012-d_S145_L007_R1_001_neg,Ob-1012-d_S149_L007_R1_001_neg,WT-DHA-a_S118_L007_R1_001_neg,Ob-911-c_S139_L007_R1_001_neg,Ob-d_S146_L007_R1_001_neg,WT-EPA-b_S127_L007_R1_001_neg,Ob-911-a_S124_L007_R1_001_neg,Ob-b_S131_L007_R1_001_neg,WT-1012-b_S130_L007_R1_001_neg,Ob-EPA-c_S137_L007_R1_001_neg,Ob-EPA-e_S153_L007_R1_001_neg,Ob-1012-b_S133_L007_R1_001_neg,WT-911-b_S129_L007_R1_001_neg,WT-911-a_S119_L007_R1_001_neg,WT-DHA-c_S134_L007_R1_001_neg,WT-EPA-d_S142_L007_R1_001_neg,Ob-a_S121_L007_R1_001_neg,Ob-DHA-c_S138_L007_R1_001_neg,WT-1012-a_S120_L007_R1_001_neg,Ob-1012-a_S125_L007_R1_001_neg,Ob-DHA-a_S123_L007_R1_001_neg,WT-e_S150_L007_R1_001_neg,Ob-911-e_S155_L007_R1_001_neg,WT-a_S116_L007_R1_001_neg,Ob-911-d_S148_L007_R1_001_neg,WT-1012-e_S152_L007_R1_001_neg,WT-911-d_S144_L007_R1_001_neg,Ob-DHA-d_S147_L007_R1_001_neg,WT-b_S126_L007_R1_001_neg
str,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,…,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64
"""10FTHFtl""",1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,…,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0
"""10FTHFtm""",1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,…,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0
"""11DOCRTSLtm""",1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,…,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0
"""11DOCRTSLtr""",1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,…,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0
"""11DOCRTSTRNtm""",1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,…,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0


In [7]:
sample = "Ob-DHA-e_S154_L007_R1_001"
combined_df.select(pl.col(sample).eq(pl.col(sample + "_neg")).alias("equal_pn").all()).collect()

equal_pn
bool
true


Ok, so yeah the pos and neg are always equal to each other.

In [4]:
df_py = pl.scan_csv("penalties_py_0.txt", separator="\t")

In [9]:
df_rs = pl.scan_csv("crates/penalties.csv")
df_rs_pos = df_rs.select(pl.col("_reaction_name").alias("Reaction") + "_pos", pl.selectors.numeric())
df_rs_neg = df_rs.select(pl.col("_reaction_name").alias("Reaction") + "_neg", pl.selectors.numeric())
df_rs = df_rs_pos.collect().extend(df_rs_neg.collect()).lazy()

In [10]:
sample = "Ob-DHA-e_S154_L007_R1_001"
df_py.join(df_rs, on="Reaction", how="inner", suffix="_rs").filter(~pl.col(sample).eq(pl.col(sample + "_rs")).alias("not equal")).select(pl.col("Reaction"), pl.col(sample), pl.col(sample + "_rs")).collect()

Reaction,Ob-DHA-e_S154_L007_R1_001,Ob-DHA-e_S154_L007_R1_001_rs
str,f64,f64
"""13DAMPPOX_pos""",0.460845,0.361041
"""2HCO3_NAt_pos""",0.160934,0.160753
"""34DHPHAMT_pos""",0.062733,0.05903
"""3AIBTm_pos""",0.120874,0.107894
"""3AIBtm_pos""",0.120874,0.107894
…,…,…
"""RE3269C_neg""",0.215986,0.162998
"""RE3308R_neg""",0.075203,0.071767
"""RE3310R_neg""",0.075203,0.071767


In [11]:
df_rs = pl.scan_csv("crates/penalties_iso.csv")
df_rs_pos = df_rs.select(pl.col("_reaction_name").alias("Reaction") + "_pos", pl.selectors.numeric())
df_rs_neg = df_rs.select(pl.col("_reaction_name").alias("Reaction") + "_neg", pl.selectors.numeric())
df_rs = df_rs_pos.collect().extend(df_rs_neg.collect()).lazy()
sample = "Ob-DHA-e_S154_L007_R1_001"
df_py.join(df_rs, on="Reaction", how="inner", suffix="_rs").filter(~pl.col(sample).eq(pl.col(sample + "_rs")).alias("not equal")).select(pl.col("Reaction"), pl.col(sample), pl.col(sample + "_rs")).collect()

Reaction,Ob-DHA-e_S154_L007_R1_001,Ob-DHA-e_S154_L007_R1_001_rs
str,f64,f64
"""5ADTSTSTERONESULT_pos""",1.0,0.158996
"""ACACT1r_pos""",0.134755,0.10781
"""ACACT1x_pos""",0.134755,0.10781
"""ADMDC_pos""",0.595922,0.207176
"""AMY1e_pos""",0.109185,0.109167
…,…,…
"""RE1796R_neg""",0.619759,0.128082
"""RE3111M_neg""",0.619759,0.128082
"""RE3111R_neg""",0.619759,0.128082


So the isoform summing change is still not quite identical. But 500 reactions differ -> 126

In [15]:
df_rs = pl.scan_csv("crates/penalties_0.csv")
df_rs_pos = df_rs.select(pl.col("_reaction_name").alias("Reaction") + "_pos", pl.selectors.numeric())
df_rs_neg = df_rs.select(pl.col("_reaction_name").alias("Reaction") + "_neg", pl.selectors.numeric())
df_rs = df_rs_pos.collect().extend(df_rs_neg.collect()).lazy()
sample = "Ob-DHA-e_S154_L007_R1_001"
print(df_py.join(df_rs, on="Reaction", how="inner", suffix="_rs").filter(pl.col(sample).sub(pl.col(sample + "_rs")).abs().gt(0.001).alias("not equal")).select(pl.col("Reaction"), pl.col(sample), pl.col(sample + "_rs")).collect())

shape: (6, 3)
┌────────────┬───────────────────────────┬──────────────────────────────┐
│ Reaction   ┆ Ob-DHA-e_S154_L007_R1_001 ┆ Ob-DHA-e_S154_L007_R1_001_rs │
│ ---        ┆ ---                       ┆ ---                          │
│ str        ┆ f64                       ┆ f64                          │
╞════════════╪═══════════════════════════╪══════════════════════════════╡
│ ENO_pos    ┆ 0.094888                  ┆ 0.30424                      │
│ PNTEH_pos  ┆ 0.10918                   ┆ 0.102644                     │
│ r0739_pos  ┆ 0.090038                  ┆ 0.086771                     │
│ CYOOm3_pos ┆ 0.116579                  ┆ 0.113491                     │
│ ENO_neg    ┆ 0.094888                  ┆ 0.30424                      │
│ r0739_neg  ┆ 0.090038                  ┆ 0.086771                     │
└────────────┴───────────────────────────┴──────────────────────────────┘


So it looks like some values here may just be floating point rounding? ATPS4m_pos 0.057161 vs 0.057163. But ENO with 0.094 vs 0.304 is noticeably different.

In [6]:
df_rs = pl.scan_csv("crates/penalties_1.csv")
df_rs_pos = df_rs.select(pl.col("_reaction_name").alias("Reaction") + "_pos", pl.selectors.numeric())
df_rs_neg = df_rs.select(pl.col("_reaction_name").alias("Reaction") + "_neg", pl.selectors.numeric())
df_rs = df_rs_pos.collect().extend(df_rs_neg.collect()).lazy()
sample = "Ob-DHA-e_S154_L007_R1_001"
print(df_py.join(df_rs, on="Reaction", how="inner", suffix="_rs").filter(pl.col(sample).sub(pl.col(sample + "_rs")).abs().gt(0.0001).alias("not equal")).select(pl.col("Reaction"), pl.col(sample), pl.col(sample + "_rs")).collect())

shape: (10, 3)
┌───────────────┬───────────────────────────┬──────────────────────────────┐
│ Reaction      ┆ Ob-DHA-e_S154_L007_R1_001 ┆ Ob-DHA-e_S154_L007_R1_001_rs │
│ ---           ┆ ---                       ┆ ---                          │
│ str           ┆ f64                       ┆ f64                          │
╞═══════════════╪═══════════════════════════╪══════════════════════════════╡
│ FAS100COA_pos ┆ 0.096286                  ┆ 0.095374                     │
│ FAS120COA_pos ┆ 0.096286                  ┆ 0.095374                     │
│ FAS140COA_pos ┆ 0.096286                  ┆ 0.095374                     │
│ FAS160COA_pos ┆ 0.096286                  ┆ 0.095374                     │
│ FAS180COA_pos ┆ 0.096286                  ┆ 0.095374                     │
│ KAS8_pos      ┆ 0.096286                  ┆ 0.095374                     │
│ PNTEH_pos     ┆ 0.10918                   ┆ 0.102644                     │
│ r0739_pos     ┆ 0.090038                  ┆ 0.086771       